blocks/api/parser.js

```diff
@@ -45,18 +78,31 @@ export function parseBlockAttributes( rawContent, attributes ) {
  * @return {Object}                All block attributes
  */
 export function getBlockAttributes( blockType, rawContent, attributes ) {
-	// Merge any attributes present in comment delimiters with those
-	// that are specified in the block implementation.
-	attributes = attributes || {};
-	if ( blockType ) {
-		attributes = {
-			...blockType.defaultAttributes,
-			...attributes,
-			...parseBlockAttributes( rawContent, blockType.attributes ),
-		};
-	}
+	const sources = mapValues( blockType.attributes, getNormalizedAttributeSource );
+
+	// Merge matcher values into attributes parsed from comment delimiters
+	attributes = {
+		...attributes,
+		...getMatcherAttributes( rawContent, sources ),
+	};
```

Comment:
Can't we avoid the `getMatcherAttributes` call and handle this while looping over `sources`?